packages/ui/src/DateTimeField.tsx

```diff
@@ -170,11 +171,22 @@ export const DateTimeField = ({
     }
   }, [value, formatValue, formattedDate, errorText]);
 
+  const iconName = useMemo(() => {
```

Comment:
This is a bit of a premature optimization. We should generally be reserving `useMemo` for "computationally expensive operations" which I don't think this is. It could just be inline or a standard variable to keep it simple.

Reply:
We can keep all the logic, just remove the `useEffect` and it'll yield the same results. So:
```
  const iconName = () => {
    if (disabled) {
      return undefined;
    } else if (type === "time") {
      return "clock";
    } else {
      return "calendar";
    }
  };
```

Reply:
You can test it by adding a few console.logs and a button to toggle disabled in the story (which I agree, should be added to the story). Because the outside prop `disabled` of DateTimeField changes, the whole component re-renders. If you do like Kyle proposed, or even simpler (below), the iconName will update as expected. And because the action sheet is a child, it also re-renders when the `disabled` prop changes.

```
  let iconName;
  if (disabled) {
    iconName = undefined;
  } else if (type === "time") {
    iconName = "clock";
  } else {
    iconName = "calendar";
  }
```

Added a branch: https://github.com/FlourishHealth/ferns-ui/tree/dateTimeField-Disabled-josh

Reply:
In React functional components, using let or const outside of a hook does not track dependencies or react to changes in state or props, which means changes in disabled won't trigger re-computation of iconName unless the entire component re-renders for another reason.
 
 When you are using the prop disabled in your branch disabled is in a hook which is triggering the rerender. This (not a reactive state to values/ props) was a persisting issue with many of our field components that i corrected in [#705](https://github.com/FlourishHealth/ferns-ui/pull/705), [706](https://github.com/FlourishHealth/ferns-ui/pull/706), 708, & 712
 

 Heres a video using the hook use memo
https://github.com/user-attachments/assets/f5959a26-d670-46ad-b400-1b0387bb1463

Heres a video with the suggested changes of let/ const

https://github.com/user-attachments/assets/721f060b-7f43-41f6-9b48-77caf9e7b7ef

Reply:
notably, because the disabled is part of dependency hook within the component that triggers the  rerender the mounting condition got DateTimeActionSheet ( `{!disabled && ...`) also gets reevealuated as well